conans/client/file_copier.py

```diff
@@ -153,7 +153,7 @@ def _filter_files(src, pattern, links, excludes, ignore_case, excluded_folders):
             filenames = {f.lower(): f for f in filenames}
             pattern = pattern.lower()
 
-        files_to_copy = fnmatch.filter(filenames, pattern)
+        files_to_copy = [n for n in filenames if fnmatch.fnmatchcase(n, pattern)]
```

Comment:
This is broken, ``filter`` is doing internally some ``os.path.normcase``, which normalizes \\ => / path in Windows too

Reply:
I am seeing this possibly breaking users, because it is by default ignoring case. 
Agree it is a bug, but as most users haven't had to specify it, I would say that the bug is in the default. The default should be ``ignore_case=True``, and then alter the behavior if it is explicitly changed to ``ignore_case=False``. Wdyt?

Reply:
Sounds good to me, indeed that's the original behavior, and changing it only after 4 years of project, probably will break users. I'll update this PR to use `ignore_case=True` by default.

Reply:
About `filter`, thanks for checking it, I'll search a better way.

Reply:
@memsharded I've split both comparisons, now for non-case-sensitive, it will keep the old behavior, as [fnmatch.filter](https://docs.python.org/3/library/fnmatch.html#fnmatch.filter) changes all file names to lower-case in Windows due [normcase](https://docs.python.org/3/library/os.path.html#os.path.normcase). For case-sensitive check, I kept same modification, but normalizing the path with [normpath](https://docs.python.org/3/library/os.path.html#os.path.normpath).